### PR TITLE
2042 kakfa plugin spark 3.2.1

### PR DESCRIPTION
**Pull request by AdrianOlosutean**

### CODE REVIEW

**Review by dk1844 — APPROVED** [review]
Hey, @AdrianOlosutean, LGTM (just read the code of the last commit that is the abris update). 

However, it apparently does not build, because on of the XMLs is not defined was not checked before. This is the fix to this branch: https://github.com/AbsaOSS/enceladus/pull/2057

**Comment by AdrianOlosutean** [comment]
> Hey, @AdrianOlosutean, LGTM (just read the code of the last commit that is the abris update).
> 
> However, it apparently does not build, because on of the XMLs is not defined was not checked before. This is the fix to this branch: #2057

Thanks, merged

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala [hunk 1]
```diff
@@ -52,20 +54,26 @@ trait CommonJobExecution extends ProjectMetadata {
                                          performance: PerformanceMeasurer)
 
   TimeZoneNormalizer.normalizeJVMTimeZone()
-  SparkVersionGuard.fromDefaultSparkCompatibilitySettings.ensureSparkVersionCompatibility(SPARK_VERSION)
+  SparkVersionGuard.fromSpark3XCompatibilitySettings.ensureSparkVersionCompatibility(SPARK_VERSION)
```
Comment (benedeki) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:57]: Don't we want to ensure the Spark is at least 3.2.1?

File: spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/HyperConformance.scala [hunk 2]
```diff
@@ -126,7 +126,7 @@ object HyperConformance extends StreamTransformerFactory with HyperConformanceAt
   override def apply(conf: Configuration): StreamTransformer = {
     log.info("Building HyperConformance")
 
-    SparkVersionGuard.fromDefaultSparkCompatibilitySettings.ensureSparkVersionCompatibility(SPARK_VERSION)
+    SparkVersionGuard.fromSpark3XCompatibilitySettings.ensureSparkVersionCompatibility(SPARK_VERSION)
```
Comment (benedeki) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/HyperConformance.scala:129]: Same - Spark 3.2.1?

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala (hunk 1 above)
Comment (AdrianOlosutean) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:57]: We can do that, but I changed to using the `SparkVersionGuard` from spark-commons, as I noticed we still had one in Enceladus. `fromSpark3XCompatibilitySettings` has 3.0.0-4.0.0 boundaries . We can instantiate a `SparkVersionGuard` manually if it makes sense

**Review by benedeki — APPROVED** [review]
- [X] code reviewed
- [x] pulled
- [x] built
- [x] run

**Review by benedeki — APPROVED** [review]
- [X] code reviewed
- [x] pulled
- [x] built
- [x] run

**Pull request merged**